File: Medias.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace AutomatizarPruebasUnitarias { 
     class Medias {

        /**
         * Calcula y regresa la media artimética
         */
         public static double mediaAritmetica(int[] vals) {
          int suma = 0;
          int resultado = 0;
          for(int i = 0; i <= vals.Length; i++) {
            suma += vals[i];
          }
          return resultado = suma/vals.Length;
        }

        /**
         * Calcula y regresa la raíz enésima = x^(1/n)
         */
        private static double raizEnesima(double x, int n) {
          return Math.Pow(x, (1 / n));
        }

        /**
         *  Usa raizEnesima para calcular y regresar la media geométrica
         */
         public double mediaGeometrica(int[] vals) {
          int suma = 0;
          for(int i=0; i < vals.Length; i++) {
            suma += 1/vals[i];
          }
          return vals.Length/suma;
        }

        /**
         * Este método no está implementado
         */
         public static double mediaArmonica(params int[] vals) {return 0; }
     }
  }

File: Medias.js
class Medias {

    mediaAritmetica(vals) {
        try {
            let suma = 0;
            if(vals === null)
                return 0.0000;
            else
                vals.forEach(element => {
                suma += element;
                }); 
                return (suma/vals.length).toFixed(4);
        }
        catch(err) {
            throw new Error(err);
        }
    }

    raizEnesima(x, n) {
        try {
            return Math.pow(x,(1/n));
        }
        catch(err) {
            throw new Error(err);
        }
    }

    mediaGeometrica(vals) {
        try {
            let mult = 1;
            if(vals === null)
                return 0.0000;
            let el = vals.filter(numero => numero === 0)
            if(el.length == vals.length && vals.length > 1)
                    throw new Error('Exception');
            else 
                vals.forEach(element => {
                mult *= element;
                });
                return this.raizEnesima(mult, vals.length).toFixed(4);
        }
        catch(err) {
            throw new Error(err);
        }
    };

    mediaArmonica (vals){
        /*let suma = 0;
        vals.forEach(element => {
            suma += (1/element);
        });
        suma = (vals.length/result);
        return suma;*/
        return "Método no implementado";
    }
}

module.exports = Medias;



